Allow CALC_ prefix as an alias for DP_

taldcroft opened the pull request:
## Description

MAUDE uses the prefix `CALC_` to indicate pseudo-MSIDs that are calculated from actual spacecraft MSIDs, for example `CALC_ROLL_FSS`. Cheta uses `DP_` for the same meaning, and many calcs have the same name apart from CALC_ vs DP_. The good news is that MAUDE also accepts `DP_` as a prefix.

This PR does a simple translation of `CALC_` to `DP_` in input MSIDs to enable interoperabiity. This is especially useful for a query that uses both CXC and MAUDE for the backend data source.

*This requires https://github.com/sot/maude/pull/39*

To do:

- [ ] Update docs

## Testing

- [x] Passes unit tests on MacOS (with new unit test)
- [n/a] Functional testing

javierggt reviewed (commented):
Looks simple enough to me.

javierggt commented on Ska/engarchive/tests/test_fetch.py:
```diff
@@ -123,6 +123,15 @@ def test_filter_bad_times_default_copy():
+    dat3 = fetch.Msid('pitch', '2020:001', '2020:002')
+    for d1, d2 in ((dat1, dat2), (dat1, dat3)):
+        assert np.all(d1.times == d2.times)
+        assert np.all(d1.vals == d2.vals)
```
I have no issues with it changing as long as one knows and perhaps documented, I just checked that it does change:
```
In [1]: from cheta import fetch

In [2]: dat1 = fetch.Msid('calc_pitch', '2020:001', '2020:002')

In [3]: dat1.msid
Out[3]: 'DP_pitch'
```

javierggt replied:
if one is going to change it, perhaps it should be "normalized", meaning that the name is always the same for a given msid.

taldcroft commented:
@javierggt - I think I addressed your concerns and made this a little better. The input msid is always left unchanged (when there is no fileglob `*`), and this user value is reflected in the plot title.

taldcroft commented:
@javierggt @jeanconn - ping that this is ready for review and I addressed the initial comments.
